app/screens/edit_server/form.tsx

```diff
@@ -0,0 +1,194 @@
+import {useIntl} from 'react-intl';
+import {Keyboard, Platform, useWindowDimensions, View} from 'react-native';
+import Button from 'react-native-button';
+import {KeyboardAwareScrollView} from 'react-native-keyboard-aware-scroll-view';
+
+import FloatingTextInput, {FloatingTextInputRef} from '@components/floating_text_input_label';
+import FormattedText from '@components/formatted_text';
+import Loading from '@components/loading';
+import {useIsTablet} from '@hooks/device';
+import {t} from '@i18n';
+import {buttonBackgroundStyle, buttonTextStyle} from '@utils/buttonStyles';
+import {changeOpacity, makeStyleSheetFromTheme} from '@utils/theme';
+import {typography} from '@utils/typography';
+import {removeProtocol, stripTrailingSlashes} from '@utils/url';
+
+type Props = {
+    buttonDisabled: boolean;
+    connecting: boolean;
+    displayName?: string;
+    displayNameError?: string;
+    handleUpdate: () => void;
+    handleDisplayNameTextChanged: (text: string) => void;
+    keyboardAwareRef: MutableRefObject<KeyboardAwareScrollView | undefined>;
+    serverUrl: string;
+    theme: Theme;
+};
+
+const getStyleSheet = makeStyleSheetFromTheme((theme: Theme) => ({
+    formContainer: {
+        alignItems: 'center',
+        maxWidth: 600,
+        width: '100%',
+        paddingHorizontal: 20,
+    },
+    fullWidth: {
+        width: '100%',
+    },
+    error: {
+        marginBottom: 18,
+    },
+    chooseText: {
+        alignSelf: 'flex-start',
+        color: changeOpacity(theme.centerChannelColor, 0.64),
+        marginTop: 8,
+        ...typography('Body', 75, 'Regular'),
+    },
+    connectButton: {
+        backgroundColor: changeOpacity(theme.centerChannelColor, 0.08),
+        width: '100%',
+        marginTop: 32,
+        marginLeft: 20,
+        marginRight: 20,
+        padding: 15,
+    },
+    loadingContainerStyle: {
+        marginRight: 10,
+        padding: 0,
+        top: -2,
+    },
+    loading: {
+        height: 20,
+        width: 20,
+    },
+}));
+
+const EditServerForm = ({
+    buttonDisabled,
+    connecting,
+    displayName = '',
+    displayNameError,
+    handleUpdate,
+    handleDisplayNameTextChanged,
+    keyboardAwareRef,
+    serverUrl,
+    theme,
+}: Props) => {
+    const {formatMessage} = useIntl();
+    const isTablet = useIsTablet();
+    const dimensions = useWindowDimensions();
+    const displayNameRef = useRef<FloatingTextInputRef>(null);
+    const styles = getStyleSheet(theme);
+
+    const onBlur = useCallback(() => {
+        if (Platform.OS === 'ios') {
+            const reset = !displayNameRef.current?.isFocused();
+            if (reset) {
+                keyboardAwareRef.current?.scrollToPosition(0, 0);
+            }
+        }
+    }, []);
+
+    const onUpdate = useCallback(() => {
+        Keyboard.dismiss();
+        handleUpdate();
+    }, [handleUpdate]);
+
+    const onFocus = useCallback(() => {
+        if (Platform.OS === 'ios') {
```

Comment:
Platform specific code can be cryptic while porting....it can be helpful to add a comment as to why we are only checking for iOS here.